File: pmd/java/listas/listaex01/ex4/Main.java
package listas.listaex01.ex4;

import java.util.List;
import java.util.Scanner;
import java.text.SimpleDateFormat;
import java.util.ArrayList;

public class Main {

  private List<Aluno> alunos;
  private List<Disciplina> disciplinas;
  private List<Matricula> matriculas;

  public Main() {
    alunos = new ArrayList<>();
    disciplinas = new ArrayList<>();
    matriculas = new ArrayList<>();
  }

  public void addAluno(Aluno aluno) {
    alunos.add(aluno);
  }

  public void addDisciplina(Disciplina disciplina) {
    disciplinas.add(disciplina);
  }

  public void addMatricula(Matricula matricula) {
    matriculas.add(matricula);
  }

  public Aluno buscarAluno(int registro) {
    for (Aluno a : alunos)
      if (a.getRegistro() == registro)
        return a;
    return null;
  }

  public Disciplina buscarDisciplina(int codigo) {
    for (Disciplina d : disciplinas)
      if (d.getCodigo() == codigo)
        return d;
    return null;
  }

  public Matricula buscarMatricula(int numero) {
    for (Matricula m : matriculas)
      if (m.getNumero() == numero)
        return m;
    return null;
  }

  public static void main(String[] args) {

    Main main = new Main();
    Matricula m = new Matricula();
    Scanner scn = new Scanner(System.in);

    while (true) {

      System.out.print(
          "\n0 - SAIR" +
              "\n1 - CADASTRAR ALUNO" +
              "\n2 - CADASTRAR DISCIPLINA" +
              "\n3 - REALIZAR MATRICULA" +
              "\n4 - VISUALIZAR MATRICULA" +
              "\nINFORME A SUA OPÇÃO: ");

      int op = scn.nextInt();

      switch (op) {

        case 0:
          scn.close();
          return;

        case 1:
          System.out.print("\nINFORME O REGISTRO E O NOME DO ALUNO: ");
          main.addAluno(
              new Aluno(
                  scn.nextInt(),
                  scn.next()));
          break;

        case 2:
          System.out.print("\nINFORME O CÓDIGO, NOME E VALOR DA DISCIPLINA: ");
          main.addDisciplina(
              new Disciplina(
                  scn.nextInt(),
                  scn.next(),
                  scn.nextFloat()));
          break;

        case 3:

          m = new Matricula();

          System.out.print("\nINFORME O NÚMERO DA MATRÍCULA: ");
          m.setNumero(scn.nextInt());

          Aluno a = null;
          while (a == null) {
            System.out.print("\nINFORME O REGISTRO DO ALUNO: ");
            int registro = scn.nextInt();
            a = main.buscarAluno(registro);
            if (a == null)
              System.out.println("\nNÃO EXISTE ALUNO COM O REGISTRO " + registro);
            else
              m.setAluno(a);
          }

          while (true) {
            System.out.print("\nINFORME O CODIGO DA DISCIPLINA OU 0 PARA PARAR: ");
            int codigo = scn.nextInt();
            if (codigo == 0)
              break;
            Disciplina d = main.buscarDisciplina(codigo);
            if (d == null)
              System.out.println("\nNÃO EXISTE DISCIPLINA COM O CÓDIGO " + codigo);
            else
              m.getDisciplinas().add(d);
          }

          main.addMatricula(m);
          break;

        case 4:
          System.out.print("\nINFORME O NÚMERO DA MATRÍCULA: ");
          int numero = scn.nextInt();
          m = main.buscarMatricula(numero);
          if (m == null)
            System.out.println("\nNÃO EXISTE MATRÍCULA COM O NÚMERO " + numero);
          else
            System.out.println(m);
      }
    }
  }

}

File: pmd/python/listas/listaex01/ex4/main.py
from datetime import date

from aluno import Aluno
from disciplina import Disciplina
from matricula import Matricula


class Main:

    def __init__(self):
        self.alunos = []
        self.disciplinas = []
        self.matriculas = []
    
    def cadastrarAluno(self):
        r = len(self.alunos) + 1
        n = input('Informe o nome do novo aluno: ')
        self.alunos.append( Aluno(r,n) )
        
    def cadastrarDisciplina(self):
        c = len(self.disciplinas) + 1
        n = input('Informe o nome da nova disciplina: ')
        v = float(input('Informe o valor da nova disciplina: '))
        self.disciplinas.append( Disciplina(c,n,v) )
        
    def cadastrarMatricula(self):
        n = len(self.matriculas) + 1
        d = date.today()
        a = self.buscarAluno(int(input('Informe o registro do aluno: ')))
        ds = []
        while True:
            cd = int(input('Informe o codigo da disciplina: '))
            ds.append(self.buscarDisciplina(cd))
            if input('Deseja incluir outra disciplina na matricula (s/n):') != 's': 
                break
        self.matriculas.append( Matricula(n,d,a,ds) )

    def buscarAluno(self, registro):
        return self.alunos[ (registro - 1) % len(self.alunos) ]
    
    def buscarDisciplina(self, codigo):
        return self.disciplinas[ (codigo - 1) % len(self.disciplinas) ]

    def buscarMatricula(self, numero):
        return self.matriculas[ (numero - 1) % len(self.matriculas) ]
    
    def main(self):
        while True:
            menu =  '\n0 - Sair\n1 - Cadastrar aluno' + \
                    '\n2 - Cadastrar disciplina\n3 - Cadastrar matricula' + \
                    '\n4 - Visualizar matricula\nDigite sua opcao: '

            op = int(input(menu))

            if op == 0:
                return
            
            elif op == 1:
                self.cadastrarAluno()
                print('\nAluno cadastrado com sucesso.\n\n') 
                
            elif op == 2:
                self.cadastrarDisciplina()
                print('\nDisciplina cadastrada com sucesso.\n\n') 
                                
            elif op == 3:
                self.cadastrarMatricula()
                print('\nMatricula cadastrada com sucesso.\n\n') 
                                
            elif op == 4:
                m = self.buscarMatricula(int(input('Informe o numero da matricula: ')))
                print('\n%s\n\n' % m) 
                                
            else:
                print('\nOpcao invalida.\n\n')        


Main().main()